Flagsmith.FlagsmithClient/Flag.cs

```diff
@@ -0,0 +1,86 @@
+using Newtonsoft.Json;
+using FlagsmithEngine.Feature.Models;
+using FlagsmithEngine.Identity.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using System.Reflection;
+using System;
+
+namespace Flagsmith
+{
+    [JsonConverter(typeof(FlagJsonConverter))]
+    public class Flag
+    {
+        public Flag() { }
+        public Flag(string name, bool enabled, string value, int featureId = default)
+        {
+            this.Enabled = enabled;
+            this.Value = value;
+            this.Name = name;
+            this.FeatureId = featureId;
+
+        }
+        [JsonProperty("id")]
+        public int Id { get; private set; }
+
+        [JsonProperty("feature.id")]
+        public int FeatureId { get; private set; }
+
+        [JsonProperty("feature.name")]
+        public string Name { get; private set; }
+
+        [JsonProperty("enabled")]
+        public bool Enabled { get; private set; }
+
+        [JsonProperty("feature_state_value")]
+        public string Value { get; private set; }
+
+        public override string ToString()
+        {
+            return JsonConvert.SerializeObject(this);
+        }
+    }
+    internal class FlagJsonConverter : JsonConverter
+    {
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
```

Comment:
Can we add some documentation to explain what this is doing?

Reply:
Can you elaborate a bit more? This method is not very readable, perhaps because I don't know .NET very well. I can't follow what it's doing and I _think_ I know what it's doing (somehow it's converting a returned API flag to a Flag object and flattening the `feature` attribute to set the `featureName` attribute?). I'm surprised that something that simple takes this much code but if it definitely does, I'd like to try and document it so that future developers can understand it's purpose. Perhaps some in line comments would help here?

Reply:
added inline comments along with a reference link.

Reply:
As discussed in the standup this morning, let's return to having the separate `Feature` class and add some `getFeatureId()` and `getFeatureName()` methods to the `Flag` class.

Reply:
as discussed removed converter and updated necessities